library/ssl_cookie.c

```diff
@@ -205,6 +297,40 @@ int mbedtls_ssl_cookie_check( void *p_ctx,
     if( cookie_len != COOKIE_LEN )
         return( -1 );
 
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+    status = psa_mac_verify_setup( &operation, ctx->psa_hmac,
+                                   ctx->psa_hmac_alg );
+    if( status != PSA_SUCCESS )
```

Comment:
Minor suggestion - maybe we could locally introduce:
```
#define PSA_ASSERT( TEST )                           \
do {                                                 \
   if( ! (TEST) )                                    \
   {                                                 \
	  ret = psa_ssl_status_to_mbedtls( status ); \
	  goto exit;                                 \
   }                                                 \
} while( 0 )
```
, or even add `== PSA_SUCCESS` on top of that. This would reduce this boilerplate by 33 lines, but I'm not sure if it's in line with PSA coding guidelines.

Reply:
I've pushed https://github.com/ARMmbed/mbedtls/blob/f4042f076ba294a87006874466fc969c0b787a1e/library/constant_time.c#L490-L495 but not sure if I should only introduce this in ssl_cookie.c and break code style with other ssl_*.c files.

Reply:
I think we want to keep things uniform. For now that probably means not introducing a macro here, but perhaps in the future we should introduce it in `ssl_misc.h` or `psa_util.h` and then use it in all (SSL) files.